src/scripts/browserDeviceProfile.js

```diff
@@ -1125,6 +1125,24 @@ export default function (options) {
         Conditions: h264CodecProfileConditions
     });
 
+    if (browser.web0s && supportsDolbyVision(options)) {
```

Comment:
At some point we probably need to make `supportsDolbyVision` (and others) return a list of containers and codecs in which DOVI is supported.
Boolean would mean "any container and codec".

Reply:
Hmm, I'm not able to fully grasp the implications of supportsDolbyVision working that way. We would push a codecprofile for each container that supportsDolbyVision returns as being okay, instead of the current logic where we are removing whatever is not supported?

Reply:
This way we can specify containers (and probably codecs) that support DOVI from the apps (we can check the TV model if it matters). And we could still probably use the trick you used.

For example:
```
supportsDolbyVision -> [] | true
append DOVI to the "base" codec profile:
    '': '...|DOVI|...'
```
```
supportsDolbyVision -> false | null | undefined
don't append DOVI to any codec profile
```
```
supportsHdr10 -> [] | true
append HDR10 to the "base" codec profile:
    '': '...|HDR10|...'
```
```
supportsDolbyVision -> ['mp4', 'mkv']
append DOVI to the "base" codec profile:
    '': '...|DOVI|...'
add codec profile:
    '-mp4,mkv': 'SDR|HLG|HDR10'
```
```
supportsHdr10 -> ['mp4', 'ts']
append HDR10 to the "base" codec profile:
    '': '...|HDR10|...'
add codec profile:
    '-mp4,ts': 'SDR|HLG|DOVI'
```

FYI, we can split the conditions into multiple codec profiles because, as I said, they will all be tested regardless of order.

At the same time, this approach looks complicated/tricky.

Reply:
I see. I'm thinking we could consider making this when another browser comes around with it's own set of conditions for a scenario similar to this, so in that case having a routine to build their specific profiles make sense. Are there any other unrelated issues or functions where this mode of function would have helped?

I'm reading posts from like 2years ago about how nvidia shield also has this problem, but I'm not sure where they are at with this now.

Reply:
> I'm thinking we could consider making this when another browser comes around with it's own set of conditions for a scenario similar to this, so in that case having a routine to build their specific profiles make sense.

Agree. I didn't say that it should be done right now. Moreover, this idea wasn't discussed with team.

> Are there any other unrelated issues or functions where this mode of function would have helped?

I don't think so. The only purpose of this form is to circumvent inconsistent support.